receiver/postgresqlreceiver/client.go

```diff
@@ -17,24 +17,23 @@ package postgresqlreceiver // import "github.com/open-telemetry/opentelemetry-co
 import (
 	"context"
 	"database/sql"
-	"errors"
 	"fmt"
 	"net"
 	"strings"
 
 	"github.com/lib/pq"
 	"go.opentelemetry.io/collector/config/confignet"
 	"go.opentelemetry.io/collector/config/configtls"
-	"go.opentelemetry.io/collector/receiver/scrapererror"
+	"go.uber.org/multierr"
 )
 
 type client interface {
 	Close() error
-	getCommitsAndRollbacks(ctx context.Context, databases []string) ([]MetricStat, error)
-	getBackends(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseSize(ctx context.Context, databases []string) ([]MetricStat, error)
-	getDatabaseTableMetrics(ctx context.Context) ([]MetricStat, error)
-	getBlocksReadByTable(ctx context.Context) ([]MetricStat, error)
+	getDatabaseStats(ctx context.Context, databases []string) (map[string]databaseStats, error)
```

Comment:
At a high level, the purpose of this PR is to address the way we represent data. I don't quite understand why that necessarily has anything to do with retrieving the data. Is it possible there are unrelated changes here? If not, what's the motivation for reworking the client in this PR?

Reply:
The main motivation is that `EmitForResource()` is order dependent so we need a way to consistently record datapoints that belong to the correct resource in an organized manner. 

For example if you collect all the database size datapoints you cannot emit them by iterating over that `MetricStat` slice because it would contain database size datapoints for both `database1`, `database2`, etc. This motivated the client to provide lookup tables so we can refer to the same database without having to go through another lookup/reducing to organize the datapoints.